circuits/src/external/blake2b.rs

```diff
@@ -49,3 +49,196 @@ pub fn hash<F: PrimeField>(
     Ok(unsafe_convert_to_bytes(layouter, native_gadget, output)?.try_into().unwrap())
 }
 
+#[cfg(test)]
+impl<F: PrimeField> FromScratch<F> for Blake2bChip<F> {
+    type Config = Blake2bConfig;
+
+    fn new_from_scratch(config: &Self::Config) -> Self {
+        Blake2bChip::new(config)
+    }
+
+    fn configure_from_scratch(
+        meta: &mut ConstraintSystem<F>,
+        instance_columns: &[Column<Instance>; 2],
+    ) -> Self::Config {
+        let advice_cols =
+            (0..NB_BLAKE2B_ADVICE_COLS).map(|_| meta.advice_column()).collect::<Vec<_>>();
+        let constant_column = meta.fixed_column();
+
+        Blake2bChip::configure(
+            meta,
+            constant_column,
+            advice_cols[0],
+            advice_cols[1..NB_BLAKE2B_ADVICE_COLS].try_into().unwrap(),
+            instance_columns[1],
+        )
+    }
+
+    fn load_from_scratch(&self, layouter: &mut impl Layouter<F>) -> Result<(), Error> {
+        self.load(layouter)
+    }
+}
+
+// A small subset of the preimage tests from the Blake2b repo, applied to our
```

Comment:
I wouldn't duplicate tests here. Instead, in this library I would test the blake2b circuits library against a blake2b CPU implementation.

This is not the most efficient one, but good enough for testing, and we already use SHA256/SHA512 with those same traits. 
https://github.com/RustCrypto/hashes